jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/RemoteBaseCatalog.java

```diff
@@ -0,0 +1,216 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.xml;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+import javax.xml.catalog.Catalog;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.jetty.util.TypeUtil;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.EntityResolver;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * <p>
+ * A catalog implementation where the {@code xml:base} is defined externally
+ * of the catalog XML, allowing for runtime determination of the {@code xml:base}
+ * (such as pointing to the contents of a remote location)
+ * </p>
+ * <p>
+ * This is a temporary Catalog implementation, and should be removed once
+ * all of our usages of {@code servlet-api-<ver>.jar} have their own
+ * {@code catalog.xml} files.
+ * </p>
+ */
+public class RemoteBaseCatalog implements Catalog, EntityResolver
```

Comment:
I like the class... but not the name.
How about `BaseClassCatalog`?

Reply:
Done